File: app/src/main/java/com/app/ecoleta/service/RetrofitService.java
package com.app.ecoleta.service;

import com.app.ecoleta.model.Cidades;
import com.app.ecoleta.model.Estados;

import java.util.List;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface RetrofitService {

    Retrofit retrofit = new Retrofit.Builder()
            .baseUrl("https://servicodados.ibge.gov.br/")
            .addConverterFactory(GsonConverterFactory.create())
            .build();

    @GET("api/v1/localidades/estados")
    Call<List<Estados>> recuperarEstados();

    @GET("api/v1/localidades/estados/{UF}/municipios")
    Call<List<Cidades>> recuperarCidades(@Path("UF") int id_uf);
}

File: app/src/main/java/com/app/ecoleta/service/RetrofitService.kt
package com.app.ecoleta.service

import com.app.ecoleta.model.Cidades
import com.app.ecoleta.model.Estados
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Path

interface RetrofitService {

    companion object {
        val retrofit = Retrofit.Builder()
            .baseUrl("https://servicodados.ibge.gov.br/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

    @GET("api/v1/localidades/estados")
    fun recuperarEstados(): Call<List<Estados?>?>?

    @GET("api/v1/localidades/estados/{UF}/municipios")
    fun recuperarCidades(@Path("UF") id_uf: Int): Call<List<Cidades?>?>?

}